### PR TITLE
Sync 1.14 branch with develop

### PR DESCRIPTION
**Pull request by jhendersonHDF**
Synchronize 1.14 branch with changes in develop after commit ea1714b3035b9de6cac885508c9f4d882c817b3f

### CODE REVIEW

**Comment by jhendersonHDF** [comment]
Please let me know if there are any changes that should not be in this PR. Due to their size, I have left out the commits for 28d2b6771f "HDF5 API test updates" and fc88fcde10 "Develop merge examples" / ac650fa9d1 "Update examples with 1.6 api and static libs". These will be merged separately and individually after this PR is merged.

File: tools/test/h5copy/CMakeTests.cmake [hunk 1]
```diff
@@ -151,12 +163,18 @@
         COMMAND ${CMAKE_CROSSCOMPILING_EMULATOR} $<TARGET_FILE:h5copy${tgt_file_ext}> -i ./testfiles/${infile} -o ./testfiles/${testname}.out.h5 -v -s ${psparam} -d ${pdparam}
     )
     set_tests_properties (H5COPY-${testname}-prefill PROPERTIES DEPENDS H5COPY-${testname}-clear-objects)
+    if ("" MATCHES "${HDF5_DISABLE_TESTS_REGEX}")
```
Comment (byrnHDF) [inline tools/test/h5copy/CMakeTests.cmake:166]: Need to fix this in develop.
S.B.
    set_tests_properties (H5COPY-${testname}-prefill PROPERTIES DEPENDS H5COPY-${testname}-clear-objects)
    if ("H5COPY-${testname}-prefill" MATCHES "${HDF5_DISABLE_TESTS_REGEX}")
      set_tests_properties (H5COPY-${testname}-prefill PROPERTIES DISABLED true)
    endif ()

**Review by byrnHDF — APPROVED** [review]
CI, CMake, Tools good. Except for fix needed in h5copy in develop.

File: src/H5Tinit_float.c [hunk 2]
```diff
@@ -51,19 +51,23 @@
  * Function:    DETECT_F
  *
  * Purpose:     This macro takes a floating point type like `double' and
- *              a base name like `natd' and detects byte order, mantissa
- *              location, exponent location, sign bit location, presence or
- *              absence of implicit mantissa bit, and exponent bias and
- *              initializes a detected_t structure with those properties.
+ *              and detects byte order, mantissa location, exponent location,
```
Comment (lrknox) [inline src/H5Tinit_float.c]:
```suggestion
 *              detects byte order, mantissa location, exponent location,
```

**Review by lrknox — APPROVED** [review]

**Pull request merged**